receiver/mongodbreceiver/scraper.go

```diff
@@ -90,9 +90,27 @@ func (s *mongodbScraper) collectMetrics(ctx context.Context, errors scrapererror
 
 	now := pcommon.NewTimestampFromTime(time.Now())
 
+	s.mb.RecordMongodbDatabaseCountDataPoint(now, int64(len(dbNames)))
 	s.collectAdminDatabase(ctx, now, errors)
-	for _, dbName := range dbNames {
-		s.collectDatabase(ctx, now, dbName, errors)
+	s.collectTopStats(ctx, now, errors)
+
+	// Mongo version 4.0+ is required to have authorized access to list collection names
+	// reference: https://www.mongodb.com/docs/manual/reference/method/db.getCollectionNames/
+	// TODO:
```

Comment:
This needs to be removed, but also - was there something that still needed to be done?
```suggestion
```

Reply:
For context, the get collection names was added in 3.0, the index stats was added in 3.2 and the option to get collection names with authorized collections was added in 4.0. As of right now [4.2](https://endoflife.date/mongodb) is the oldest version currently being supported but we have earlier versions 2.6 and 3.0 being supported in the readme. So between 3.2-4.0 `index.access.count` would not be collected. If the mongodb version check were changed from 3.0 to 4.0 then an access error would be produced if not ran in admin mode. I am not really sure where to document this.

Reply:
Sounds like we _actually_ support version 4.2+? Is this a change, or a just a documentation issue?

Reply:
It sounds like the older versions are still supported, but these new metrics are only available in more recent versions. If that's the case, I think we should make sure that's in the documentation for those metrics

Reply:
I agree we should document clearly any metrics that are version dependent.

However, I have to question whether we _should_ be supporting versions that are years past EOL. If we're going to drop support, it should happen before the component is marked stable. WDYT?

Reply:
I believe the reason this receiver supporting such a huge range of old versions is there were concerns about newer versions not being available in package managers. I'll check with the team who made that call.

Reply:
Confirmed that at least in December 2021, 2.x and 3.x were still the defaults for some distros
<img width="285" alt="image" src="https://user-images.githubusercontent.com/1297166/181834312-1688789f-35b3-4b69-8933-94575022191f.png">